platforms/core-runtime/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/AbstractInstrumentationProcessor.java

```diff
@@ -168,4 +175,66 @@ private void runCodeGeneration(List<CallInterceptionRequest> requests) {
 
         generatorHost.generateCodeForRequestedInterceptors(requests);
     }
+
+    /**
+     * Discover all elements annotated with the given annotations, skipping package roots.
```

Comment:
💭 I think this comment described the issue, but I've noticed that in some cases we discover also classes from other projects using `RoundEnvironment#getElementsAnnotatedWith(Class)` and then we have duplicated generated classes which cases bunch of different issues. So this fixes that

Reply:
Great that you describe the problem here. I'd also add instructions on what to do with this once the issue is fixed.